crates/aptos-compression/src/lib.rs

```diff
@@ -108,6 +129,35 @@ pub fn decompress(
     Ok(raw_data)
 }
 
+// Derived from lz4 library
```

Comment:
we probably should attach a link here, also does compression mode changes how this is calculated? we should probably have upstream exposing the function public